File: fib.rb
def fibonacci(n)
  return n if n <= 1
  a, b = 0, 1
  (2..n).each do
    a, b = b, a + b
  end
  b
end

start_time = Process.clock_gettime(Process::CLOCK_MONOTONIC)
result = fibonacci(50)
end_time = Process.clock_gettime(Process::CLOCK_MONOTONIC)

time_taken = end_time - start_time

# Get memory usage in KB
memory_kb = `ps -o rss= -p #{Process.pid}`.to_i

puts "Language: Ruby"
puts format("Time: %.6f seconds", time_taken)
puts "Memory: #{memory_kb} KB"
puts "Result: #{result}"

File: fib.py
import time
import tracemalloc

def fibonacci(n):
    if n <= 1:
        return n
    a, b = 0, 1
    for _ in range(2, n + 1):
        a, b = b, a + b
    return b

tracemalloc.start()
start_time = time.perf_counter()
result = fibonacci(50)
end_time = time.perf_counter()
current, peak = tracemalloc.get_traced_memory()
tracemalloc.stop()

time_taken = end_time - start_time
memory_kb = peak // 1024

print("Language: Python")
print(f"Time: {time_taken:.6f} seconds")
print(f"Memory: {memory_kb} KB")
print(f"Result: {result}")
